### PR TITLE
Fixed some documentation about boundary and manifold.

### PR DESCRIPTION
**Pull request by luca-heltai**
Part of #6274

### CODE REVIEW

File: doc/doxygen/headers/manifold.h [hunk 1]
```diff
@@ -34,21 +34,23 @@
...
+ *   Triangulation::set_manifold() function associating it with a manifold id
+ *   (see types::manifold_id), use this manifold id on the cells, faces or edges
+ *   of the triangulation that should be described by this manifold using the
+ *   TriaAccessor::set_manifold_id() function, and then the Triangulation will
+ *   ask the manifold object where a new vertex to be located on a cell, face or
+ *   edge so attributed should be located upon mesh refinement. Several classes
+ *   already exist to support the most common geometries, e.g.,
+ *   CylinderManifold, or PolarManifold, which represent respectively the
+ *   geometry obtained when describing your space in cylindrical coordinates or
+ *   in polar coordinates, and by default, all curved geometries generated using
```
Comment (tjhei) [inline doc/doxygen/headers/manifold.h]: "coordinates. By default, ..."

File: doc/doxygen/headers/manifold.h [hunk 2]
```diff
@@ -34,21 +34,23 @@
...
+ *   functions in the GridGenerator namespace attach by default the correct
```
Comment (tjhei) [inline doc/doxygen/headers/manifold.h]: remove "by default"

File: doc/doxygen/headers/manifold.h [hunk 3]
```diff
@@ -34,21 +34,23 @@
...
+ *   describe the geometry of a domain. One can then attach an object of a class
+ *   derived from this base class to the Triangulation object using the
+ *   Triangulation::set_manifold() function associating it with a manifold id
+ *   (see types::manifold_id), use this manifold id on the cells, faces or edges
+ *   of the triangulation that should be described by this manifold using the
+ *   TriaAccessor::set_manifold_id() function, and then the Triangulation will
+ *   ask the manifold object where a new vertex to be located on a cell, face or
+ *   edge so attributed should be located upon mesh refinement. Several classes
+ *   already exist to support the most common geometries, e.g.,
+ *   CylinderManifold, or PolarManifold, which represent respectively the
```
Comment (tjhei) [inline doc/doxygen/headers/manifold.h]: CylindricalManifold

File: doc/doxygen/headers/boundary.h [hunk 4]
```diff
@@ -79,64 +79,53 @@
  * <tt>boundary_id</tt> that uniquely identifies which part of the
  * boundary this face is on. If nothing is specified at creation time,
  * each boundary face has a zero boundary id and each triangulation
- * object has an invalid manifold id. On the other hand, the boundary
+ * object has an flat manifold id. On the other hand, the boundary
```
Comment (jppelteret) [inline doc/doxygen/headers/boundary.h]: I guess that, to be consistent with the majority of the documentation below, this should be `manifold_id`.

(sorry for the duplicate indications later on)

File: doc/doxygen/headers/boundary.h [hunk 5]
```diff
@@ -79,64 +79,53 @@
...
- * maintained for a while, but might eventually be changed. The suggested
- * strategy is to use manifold_ids to describe the geometry, and boundary_ids
- * to describe boundary conditions.
- * 
+ * Before version 8.2, the library allowed only boundary faces to follow a
+ * curved geometric description. Since version 8.2 this has been introduced also
+ * for interior faces and cells, and the boundary_id has been separated from the
+ * manifold_id. The former is used to identify the type of boundary conditions
+ * to apply, while the latter is used to identify the geometry and describe how
+ * new vertices should be created upon refinement of the mesh, or where high
```
Comment (jppelteret) [inline doc/doxygen/headers/boundary.h]: high ->higher (?)

File: doc/doxygen/headers/boundary.h [hunk 6]
```diff
@@ -79,64 +79,53 @@
...
+ * order Mapping objects should place their support points on the exact
+ * geometry.
+ *
+ * Since version 9.0 of the library, the boundary_id associated to the boundary
+ * faces is ignored by Manifold objects, and Manifold descriptors can only be
+ * attached to manifold ids.
```
Comment (jppelteret) [inline doc/doxygen/headers/boundary.h]: manifold_id's

File: doc/doxygen/headers/boundary.h [hunk 7]
```diff
@@ -79,64 +79,53 @@
...
+ *
+ * The behavior of the Triangulation class w.r.t. geometry descriptions is the
```
Comment (jppelteret) [inline doc/doxygen/headers/boundary.h]: w.r.t. -> with respect to

(I'm not sure how many of our non-proficient/non-native English users would understand the abbreviation)

File: doc/doxygen/headers/boundary.h [hunk 8]
```diff
@@ -79,64 +79,53 @@
...
+ * following: Triangulation::set_manifold() attaches a manifold descriptor to
+ * the specified manifold id. The function expects a Manifold descriptor, and
```
Comment (jppelteret) [inline doc/doxygen/headers/boundary.h]: manifold_id

File: doc/doxygen/headers/boundary.h [hunk 9]
```diff
@@ -79,64 +79,53 @@
...
+ * you could describe both the interior and the boundary of the domain using the
+ * same object.
+ *
+ * Whenever a new vertex is needed in an object, the Triangulation queries the
+ * manifold_id of the object which needs refinement. If the query resulted in a
+ * number different from numbers::flat_manifold_id, then the Triangulation looks
+ * whether a previous call to Triangulation::set_manifold() was performed with
+ * the given id, and if yes, it uses the stored object to obtain new vertices,
```
Comment (jppelteret) [inline doc/doxygen/headers/boundary.h]: May I suggest the following to break up this long sentence:

> ... was performed with the given id. If it was, then the triangulation uses the stored object...

File: doc/doxygen/headers/boundary.h [hunk 10]
```diff
@@ -79,64 +79,53 @@
...
+ * otherwise it uses a FlatManifold object.
+ *
+ * @note This behavior is **not** backward compatible to that of deal.II
+ * versions prior to 9.0. If one ignores the manifold_id of an object (i.e., if
+ * it has never been set), by default it is and remains set to
+ * numbers::flat_manifold_id. In previous versions of the library, the
+ * boundary_id of the boundary faces would be queried and used in place of the
+ * manifold_id. If you have old programs that only set boundary ids, you should
```
Comment (jppelteret) [inline doc/doxygen/headers/boundary.h]: boundary_id's

File: doc/doxygen/headers/boundary.h [hunk 11]
```diff
@@ -79,64 +79,53 @@
...
+ * modify them to use manifold ids instead (or you could use
```
Comment (jppelteret) [inline doc/doxygen/headers/boundary.h]: manifold_id's

File: doc/doxygen/headers/manifold.h [hunk 12]
```diff
@@ -34,21 +34,23 @@
...
- *   mesh refinement. Several classes already exist to support the most common
- *   geometries, e.g., CylinderManifold, or PolarManifold, which represent
- *   respectively the geometry obtained when describing your space in
- *   cylindrical coordinates or in polar coordinates.
+ *   On the other hand, if one deals with curved geometries, or geometries which
+ *   require a denser refinement in some direction, this is not the appropriate
+ *   thing to do. The classes derived from the Manifold base class therefore
+ *   describe the geometry of a domain. One can then attach an object of a class
+ *   derived from this base class to the Triangulation object using the
+ *   Triangulation::set_manifold() function associating it with a manifold id
```
Comment (jppelteret) [inline doc/doxygen/headers/manifold.h]: manifold_id

(on the next line too)

File: doc/doxygen/headers/boundary.h (hunk 6 above)
Comment (drwells) [reply doc/doxygen/headers/boundary.h]: I believe the original version is correct (this is plural not possessive)

File: doc/doxygen/headers/boundary.h (hunk 10 above)
Comment (drwells) [reply doc/doxygen/headers/boundary.h]: same thing: if I have read this correctly then it should be plural and not possessive and there should not be an `'`

File: doc/doxygen/headers/boundary.h (hunk 11 above)
Comment (drwells) [reply doc/doxygen/headers/boundary.h]: (same)

File: doc/doxygen/headers/boundary.h (hunk 5 above)
Comment (drwells) [reply doc/doxygen/headers/boundary.h]: I believe this is fine the way it is; both work but 'high order' is an acceptable adjective phrase.

File: doc/doxygen/headers/boundary.h (hunk 6 above)
Comment (jppelteret) [reply doc/doxygen/headers/boundary.h]: Although I agree with @drwells on the plural vs possessive point, what I was trying to do was make the use of `manifold_id` consistent throughout the text. It seems strange swapping back and forth between `manifold_id` and `manifold id`, but at the same time writing it as `manifold_ids` didn't seem correct either. I also didn't want to suggest that we change this to `<tt>manifold_id<\tt>s`. Either way, if @luca-heltai would prefer to keep the original version then I wouldn't press the point any further.

**Comment by luca-heltai** [comment]
I'm neutral on this. What do others think? Would `manifold_id`s be better than manifold ids?

**Comment by drwells** [comment]
I could go either way since I don't think it generates a link.

**Comment by jppelteret** [comment]
@luca-heltai This PR is undeserving of being held up for much longer due to my pedantry, so if there are no more comments by tomorrow then lets merge this as it is. I appreciate your making the other touch-ups that I suggested.

**Comment by bangerth** [comment]
Making an executive decision here to merge these doc patches as-is. If anyone feels strongly enough about `manifold ids` vs `<code>manifold_id</code>s`, then that can be addressed in follow-up patches.

**Comment by drwells** [comment]
@jppelteret I hope that I did not come on too strong with the comment on apostrophes. Its not you, its me.

**Comment by jppelteret** [comment]
> Making an executive decision here to merge these doc patches as-is

Good, sometimes we need a tie-breaker on these non-critical issues.

>  I hope that I did not come on too strong with the comment on apostrophes. 

Not at all, I wasn't put out in the slightest :-)

**Pull request merged**